addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/NikoHomeControlDiscover.java

```diff
@@ -0,0 +1,135 @@
+ * The Niko Home Control IP-interface responds to this UDP packet.
+ * The IP-address from the Niko Home Control IP-interface is then extracted from the response packet.
+ * The data content of the response packet is used as a unique identifier for the bridge.
+ *
+ * @author Mark Herwege
+ */
+public final class NikoHomeControlDiscover {
+
+    private Logger logger = LoggerFactory.getLogger(NikoHomeControlCommunication.class);
+
+    private InetAddress addr;
+    private String nhcBridgeId;
+
+    public NikoHomeControlDiscover(InetAddress addr) throws IOException {
+
+        InetAddress broadcastaddr = addr;
+        if (broadcastaddr == null) {
+            broadcastaddr = getBroadcastAddress();
+        }
+        if (broadcastaddr == null) {
+            throw new IOException("Cannot determine broadcast address");
+        }
+
+        final byte[] discoverbuffer = { (byte) 0x44 };
+        final int broadcastport = 10000;
+
+        DatagramPacket discoveryPacket = new DatagramPacket(discoverbuffer, discoverbuffer.length, broadcastaddr,
+                broadcastport);
+        byte[] buffer = new byte[1024];
+        DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+
+        try (DatagramSocket datagramSocket = new DatagramSocket(null)) {
+            datagramSocket.setBroadcast(true);
+            datagramSocket.setSoTimeout(500);
+            datagramSocket.send(discoveryPacket);
+            datagramSocket.receive(packet);
+            this.addr = packet.getAddress();
+            setNhcBridgeId(packet);
+            logger.debug("Niko Home Control: IP address is {}, unique ID is {}", this.addr, this.nhcBridgeId);
+        }
+
+    }
+
+    public NikoHomeControlDiscover() throws IOException {
+        this(null);
+    }
+
+    /**
+     * @return the addr
+     */
+    public InetAddress getAddr() {
+        return this.addr;
+    }
+
+    /**
+     * @return the nhcBridgeId
+     */
+    public String getNhcBridgeId() {
+        return this.nhcBridgeId;
+    }
+
+    /**
+     * Retrieves a unique ID from the returned datagram packet received after sending the UDP discovery message.
+     *
+     * @param packet
+     */
+    private void setNhcBridgeId(DatagramPacket packet) {
+
+        byte[] packetData = packet.getData();
+        int packetLength = packet.getLength();
+        packetLength = packetLength > 6 ? 6 : packetLength;
+        StringBuilder sb = new StringBuilder(packetLength);
+        for (int i = 0; i < packetLength; i++) {
+            sb.append(String.format("%02x", packetData[i]));
+        }
+        this.nhcBridgeId = sb.toString();
+
+    }
+
+    /**
+     * Method to try to find a broadcast address from the local network interfaces.
+     * If this does not yield the correct result, the broadcast address needs to be set in the configuration.
+     *
+     * @return current broadcast address
+     */
+    private InetAddress getBroadcastAddress() {
```

Comment:
Yes, I think it could make sense. I would need to figure out how to do this properly though. This is binding is my first experience using git and doing any Java programming at all. Could I leave it in here for now and change the code when I have this moved over to eclipse.smarthome and accepted there?